crates/telio-firewall/src/firewall.rs

```diff
@@ -685,6 +685,27 @@ impl StatefullFirewall {
                     || it == v6::EchoReply.0
                 {
                     it
+                } else if (it == v4::DestinationUnreachable.0
+                    || it == v4::TimeExceeded.0
+                    || it == v4::ParameterProblem.0)
+                    && ip.get_next_level_protocol() == IpNextHeaderProtocols::Icmp
+                {
+                    let packet = unwrap_option_or_return!(
+                        Ipv4Packet::try_from(&icmp_packet.payload()[4..]),
+                        None
+                    );
+                    return Self::build_icmp_key(&packet, false);
```

Comment:
I believe the call to build_icmp_key(...) here will almost always return None, because the packet contained in icmp_packet.payload() is generally not gonna be ICMP one.